### PR TITLE
fix returned error values in defer functions

### LINKED ISSUE #943: bug in the third package sqlutils
I am not sure whether this question is suitable here.

I find a problem in the third package `github.com/openark/golib/sqlutils` used in `orchestrator`.

package: `github.com/openark/golib/sqlutils`
file: ` golib/sqlutils/sqlutils.go`
code:
```
// QueryRowsMap is a convenience function allowing querying a result set while poviding a callback
// function activated per read row.
func QueryRowsMap(db *sql.DB, query string, on_row func(RowMap) error, args ...interface{}) error {
	var err error
	defer func() {
		if derr := recover(); derr != nil {
			err = errors.New(fmt.Sprintf("QueryRowsMap unexpected error: %+v", derr))
		}
	}()

... ...
	return err
}
```
The `err` modified in `defer` will not return to the caller.

For example:
```
package main

import (
        "fmt"
)

func main(){
        i := test()
        fmt.Println("main i:", i)
}

func test() int {
        var a int
        defer func () {
                a = 2
        }()

        a = 1
        return a
}
```
output:
`main i: 1`

The correct usage like this:
```
package main

import (
        "fmt"
)

func main(){
        i := test()
        fmt.Println("main i:", i)
}

func test() (a int) {
        defer func () {
                a = 2
        }()

        a = 1
        return a
}
```
output:
`main i: 2`

I also submitted an issue [here](https://github.com/outbrain/golib/issues/5).

Please check it.
cc @MOON-CLJ

### PR DESCRIPTION
**Pull request by shlomi-noach**
Fixes https://github.com/github/orchestrator/issues/943 (and more scenarios like the one reported).

An `err` was not correctly returned when modified by a `defer` function in `sqlutils.go`.

### CODE REVIEW

**Review by yangeagle — APPROVED** [review]

**Review by zmoazeni — APPROVED** [review]
Hah. I was going through and creating a PR for this by adding a test file to the original repo. Beat me to the punch.

File: vendor/github.com/openark/golib/sqlutils/sqlutils.go [hunk 1]
```diff
@@ -335,20 +332,18 @@ func ExecNoPrepare(db *sql.DB, query string, args ...interface{}) (sql.Result, e
...
-	var err error
+func execInternal(silent bool, db *sql.DB, query string, args ...interface{}) (res sql.Result,err error) {
 	defer func() {
 		if derr := recover(); derr != nil {
 			err = errors.New(fmt.Sprintf("execInternal unexpected error: %+v", derr))
 		}
 	}()
-
-	stmt, err := db.Prepare(query)
+var stmt *sql.Stmt
```
Comment (zmoazeni) [inline vendor/github.com/openark/golib/sqlutils/sqlutils.go]: Oh you do have a spacing issue here.

Comment (shlomi-noach) [reply vendor/github.com/openark/golib/sqlutils/sqlutils.go]: Huh. I wonder how this passed CI? CI checks diff vs. `gofmt`-ed code.

**Comment by shlomi-noach** [comment]
>  adding a test file to the original repo

@zmoazeni _please do_

**Comment by shlomi-noach** [comment]
Sorry about the delay; I expect to look into & merge a bunch of community PRs during the first two weeks of October.

**Comment by zmoazeni** [comment]
All good @shlomi-noach. I've been super busy on my side too. Haven't done anything in OSS-land lately.

**Pull request merged**